### PR TITLE
added trouble shooting build hints

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
After several request for help on our issues, I compiled a couple of hits on how to build/rebuild onnx-mlir. Hopefully it will be helpful to others and reduce the traffic of issues on this matter.

Signed-off-by: Alexandre Eichenberger <alexe@us.ibm.com>

### CODE REVIEW

File: docs/TestingHighLevel.md [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
+
+Level used in the code is found by executing a `git log` in the `llvm-project` subdirectory.
+
+If they don't match, please update the llvm project to the required level.
+
+### 2) Checking the right third_party support
+
+Typically, when we update the ONNX op level, it results in new software in the `third_party/onnx` subdirectory. Failing to update that code results typically in compiler failures related to ONNX dialect code.
+
+I find it easier to simply remove the `third_party` directory and then reinstalling the code using `git submodule update --init --recursive`.
```
Comment (etiotto) [inline docs/TestingHighLevel.md:27]: [suggestion]: "I find it easier" -> "It is easier"

**Review by etiotto — APPROVED** [review]
LGTM

File: docs/TestingHighLevel.md [hunk 2]
```diff
@@ -1,31 +1,62 @@
...
-# ONNX-MLIR: Testing and Specific Environment variables
+# ONNX-MLIR: Build trouble-shooting and testing ONNX_MLIR
+
+## Trouble shooting the building of ONNX-MLIR
+
+If you have issues during the first `onnx-mlir` build, you may need to check the environment variables used by our build. See the last section of this page for help.
+
+If you have used the source directory successfully for a while, you may experience difficulties to rebuild `onnx-mlir` after merging the latest changes from the `main` branch.
+
+Below is a couple of steps you may perform. If any of them apply, my recommendation is to remove the `onnx-mlir/build` subdirectory and rebuild from scratch using the `cmake` commands.
```
[the remark this answers is not in the document]
Comment (AlexandreEichenberger) [reply docs/TestingHighLevel.md:11]: yes, make sense

File: docs/TestingHighLevel.md (hunk 1 above)
Comment (AlexandreEichenberger) [reply docs/TestingHighLevel.md:27]: yes, tx

**Pull request merged**